src/pages/Menu.js

```diff
@@ -1,7 +1,20 @@
 import MenuItem from "../components/MenuItem";
 
-export default function Menu({ menuItems, onUpdateQuantity: updateQuantity }) {
-  return menuItems.map((item) => {
+export default function Menu({
+  menu,
+  order,
+  onUpdateQuantity: updateQuantity,
+}) {
+  return menu.map((item) => {
+    const filteredOrderItem = order.filter((orderItem) => {
+      return orderItem.id === item.id;
+    });
+    if (filteredOrderItem.length > 0) {
+      item.quantity = filteredOrderItem[0].quantity;
+    } else {
+      item.quantity = 0;
+    }
+
     return (
       <MenuItem key={item.id} item={item} onUpdateQuantity={updateQuantity} />
     );
```

Comment:
this part is very confusing.
I think it would look much better if you put the logic before the return part and just return a function like "renderMenuItems()" or something similar.